### DISCUSSION TITLE
Random 🎲

### DISCUSSION THREAD

**Discussion by loulou64490** [Jeffser/Alpaca#149] [General]
[https://www.instagram.com/reel/C7mYj_poofx/?igsh=MWJrcTUxYTQ1Nm5idQ==](https://www.instagram.com/reel/C7mYj_poofx/?igsh=MWJrcTUxYTQ1Nm5idQ==)

**Comment by Jeffser** [comment]
🦙

**Reply by olumolu** [reply]
I think creating a mastodon acc for the project creates a good impact. @Jeffser

**Reply by Jeffser** [reply]
I believe most people on mastodon (mostly of floss.social) follow individual developers rather than accounts from projects, I could be wrong tho

**Reply by olumolu** [reply]
https://fosstodon.org/@FreeTube
https://mastodon.art/@Krita
https://mastodon.world/@signalapp
Few apps.

**Reply by Jeffser** [reply]
alright I made this profile on floss https://floss.social/@alpaca
don't expect me to post much tho

**Reply by olumolu** [reply]
You can share with someone who can do that.

**Reply by Jeffser** [reply]
nah

**Comment by olumolu** [comment]
On the front page of the project you have written gtk4 adwaita but gtk4 is is using libadwaita which is the newest themeing of gtk4 apps. @Jeffser

**Reply by Jeffser** [reply]
technically not the same, Alpaca uses Adwaita widgets

**Reply by olumolu** [reply]
Ohh

**Comment by loulou64490** [comment]
Translating take so much time :sob: :sob: 
I was on holiday and I couldn't follow the update...

**Reply by Jeffser** [reply]
take your time hahaha

**Reply by loulou64490** [reply]
u2

**Comment by loulou64490** [comment]
![20240809_225116](https://github.com/user-attachments/assets/225f8481-3ad4-41f3-b798-3d87a769fd3d)
redesigning all the website (with a gradient and a good font ) (and better web development skill lol)

**Reply by loulou64490** [reply]
![Screenshot 2024-08-10 at 00-43-39 Alpaca](https://github.com/user-attachments/assets/e4198728-47f9-4c53-bd2c-d22741f621f8)

**Reply by Jeffser** [reply]
that looks great!

**Comment by loulou64490** [comment]
I AM THE BUG GENERATOR
![Capture d’écran du 2024-08-10 18-56-34](https://github.com/user-attachments/assets/6cda8ecb-4be6-4815-a985-937f7734e4fc)

**Reply by Jeffser** [reply]
you are really making me work extra time huh

**Reply by Jeffser** [reply]
thanks for the donation btw <3

**Reply by loulou64490** [reply]
![Capture d’écran du 2024-08-11 07-37-37](https://github.com/user-attachments/assets/29805628-b690-4cea-addd-905d489d21a4)
preview :eyes:

**Reply by olumolu** [reply]
This looks really nice btw have you tested
Zed app this is faster than vs code you can try

**Reply by Jeffser** [reply]
> preview 👀

oh wow that looks great!

**Reply by loulou64490** [reply]
> This looks really nice btw have you tested
> Zed app this is faster than vs code you can try

Yes, pretty fast, but it's developed in rust so it's designed for rust, not for web development... and there isn't intellisense, Emmet abreviation, live preview on local server in one click, etc... and this IDE is too young, the stable linux version was released a month ago, there is not enough extension 
And I have a good PC so vscode is fast for me

**Reply by olumolu** [reply]
> > This looks really nice btw have you tested
> > Zed app this is faster than vs code you can try
> 
> Yes, pretty fast, but it's developed in rust so it's designed for rust, not for web development... and there isn't intellisense, Emmet abreviation, live preview on local server in one click, etc... and this IDE is too young, the stable linux version was released a month ago, there is not enough extension And I have a good PC so vscode is fast for me

No this is good for web development also it is just a drop in replacement for vs code by the atom original devlopers. Just use it. You can install this with flathub.
Extension is only an issue which they are improving.

**Reply by loulou64490** [reply]
I try it but I don't like it. maybe in a year or two i'll try again

**Reply by loulou64490** [reply]
![Screenshot 2024-08-12 at 00-49-56 Alpaca](https://github.com/user-attachments/assets/986b43d6-8238-4814-8e34-882ca839ac91)

I don't have image for YouTube transcription 😭

**Comment by olumolu** [comment]
Will you consider for macos support?

**Reply by loulou64490** [reply]
[https://www.gtk.org/docs/installations/macos](https://www.gtk.org/docs/installations/macos)

**Reply by Jeffser** [reply]
I will look into it, I just happen to have a hackintosh laying around

**Comment by loulou64490** [comment]
200 stars, wow

**Reply by olumolu** [reply]
Afford paid off.

**Comment by loulou64490** [comment]
![Screenshot 2024-12-25 12 00 06](https://github.com/user-attachments/assets/63db481a-9a84-42a6-a36a-b8926ddda88a)
work on chrome os 👍👍

**Comment by Bajahaw** [comment]
Hello guys, is there any plans for integrating tts engines e.g piper? or at least make it possible for the user to do so? I think it would be cool to have.

**Reply by Jeffser** [reply]
Hi, there's no current plan for that but I'll add it to the [project plan](https://github.com/Jeffser/Alpaca/wiki/Project-Plan) as an idea "worth looking into"

Thanks for the suggestion

**Reply by Bajahaw** [reply]
Great, thanks for the response. for me it is the only thing that keeps me from using Alpaca primarily.

**Reply by Jeffser** [reply]
Hi, Alpaca supports TTS since 6.0.0

**Comment by mags0ft** [comment]
Yippie, Alpaca hit 700 Stars on GitHub! Congratz.

**Comment by mags0ft** [comment]
WiFi didn't work today.

All the students with their iPads: *cannot access ChatGPT*

Alpaca on my Laptop:

![IMG_20250226_131127.jpg](https://github.com/user-attachments/assets/f7b4dbf8-db06-4e4e-937f-bd141422d362)

**Reply by olumolu** [reply]
Which model are you using
I have switched to deepseek completely

**Reply by mags0ft** [reply]
For me personally, on my laptop, Llama 3.2 3b works best as it tends to answer more quickly and performed better in my personal benchmarks. Sometimes, DeepSeek-R1 1.5b is great, too - but only when advanced reasoning is necessary.

On my actual machine at home, I tried out Llama 3.1 8b and EuroLLM (inside of Ollama) which both performed... reasonably well?

I figured all those models work really well together, so I made a little side project called [Jace](https://github.com/mags0ft/Jace) (sneaky product placement, I know haha) that lets those models discuss with each other in a "council". Is quite interesting to watch them do that, too!

**Reply by olumolu** [reply]
Seitch to qwen which is far better than llama models like in my tests wwen 7b is better when compared to 8b llama

**Reply by mags0ft** [reply]
I have almost switched to Gemma 3 by now. It's really strong. 27B outperforms the proprietary Gemini in my personal tasks.

**Reply by mags0ft** [reply]
Update: Mistral Small 3.1 is really flipping strong.

https://ollama.com/library/mistral-small3.1

**Reply by olumolu** [reply]
> Update: Mistral Small 3.1 is really flipping strong.
> 
> https://ollama.com/library/mistral-small3.1

you may like it but i dont think this is not the strongest model for the size and i dont like there license approach
https://artificialanalysis.ai/#artificial-analysis-intelligence-index

**Reply by mags0ft** [reply]
Haven't said it's the strongest of them, but benchmarks are looking really good. The benchmarks you've cited do not normalize for size.

And what's problematic about Apache 2.0 licensing? Seriously interested, I do not mean to sound overly questioning.

**Reply by olumolu** [reply]
my bad they provide this under apache but i read about this that they provide models under some special licenses in which you cant use on commercial space or something.

**Reply by mags0ft** [reply]
Qwen 3 is the next go-to model in my opinion. It's even stronger than the other models.

**Reply by olumolu** [reply]
True but it is really hard to run 30b model on a home system.
Btw do you know how to get hugging face api work with so i can run hugging chat models with alpaca...
I can't find or understand how it work. https://huggingface.co/chat/
I want to run hugging chat if i can get it working with curl i will add this to alpaca but hugging chat is really a free way to chat with models.

**Reply by mags0ft** [reply]
Hmm, do they even have an API for their chats? They have the inference API after all, but it differs heavily in use case and functionality. I don't think implementing support for Inference Endpoints would be reasonable.

I found [this](https://github.com/Soulter/hugging-chat-api), but it's unofficial and would mean adding another dependency. Which isn't ideal at all.

**Comment by mags0ft** [comment]
![image](https://github.com/user-attachments/assets/dc887e45-b434-49b3-b7e6-ffc6ddc37f58)

Jeffry's gonna see all these issues spawn in and cry :sob:

**Reply by Jeffser** [reply]
I appreciate the bug reports but at least he should've spaced them out by a couple of hours, my email imploded hahaha

**Comment by mags0ft** [comment]
Well, look who I found today!

![Photo of an Alpaca standing a bit farther away](https://github.com/user-attachments/assets/d3630bad-805c-4826-b812-a5b28db7ad0c)

![Closeup of an Alpaca](https://github.com/user-attachments/assets/01ae89c8-39f0-4d7f-9c48-35ef6425935a)

**Reply by mags0ft** [reply]
It couldn't tell me why my integrated Ollama instance wasn't working in 6.0.1 though...

**Reply by olumolu** [reply]
i find the app really crashing again and again i think features are enough but the implementation lacks the polishness. i think alpaca need polish when i use this i feeel like using an alpha appp

**Reply by mags0ft** [reply]
Yes, crashes are very common now but super hard to trace back. I tried investigating their origin several times, but often, there isn't even an exception thrown, just a cryptical GTK error. Hard to pinpoint what's wrong.

**Reply by Jeffser** [reply]
Most bugs have been ironed out in the newest release (6.0.3)

**Reply by mags0ft** [reply]
Perfect! 🤗

**Reply by olumolu** [reply]
I just had an app crash when it crash in mist case the app show a black window but app stay on the screen.
I think this is only happening when we use it with apis.

**Comment by mags0ft** [comment]
Alpaca's approaching the 1k stars on GitHub, woohoo! 🙌

**Comment by mags0ft** [comment]
Woo-hoo!

![Diagram depicting the start history of the Alpaca GitHub repository, approaching one thousand stars](https://github.com/user-attachments/assets/eaec9656-a627-4709-80bf-0f960aae7bd1)

**Comment by mags0ft** [comment]
![Screenshot showing a commit by user Jeffser, saying "Live message rendering (yeh really)"](https://github.com/user-attachments/assets/aef0f719-160b-43c2-b1bd-c750ea41b7ca)

Oh lord guys and gals, Jeffry is cooking!

**Reply by Jeffser** [reply]
yeah really

**Reply by Jeffser** [reply]
Exclusive sneak peek (Courtesy of my slow af computer lmao)

[Screencast From 2025-06-21 15-04-12.webm](https://github.com/user-attachments/assets/3aa5843a-809a-4b78-9f79-d60ff6394b67)

**Reply by Bajahaw** [reply]
hey, don't you think the spacing between lines can be improved? it feels like a bit squished .. 

just a suggestion .. Thank you!

**Comment by Jeffser** [comment]
[Another classic commit](https://github.com/Jeffser/Alpaca/commit/a22f843e09ca291345cc377ab9ce534f228489ca)

![image](https://github.com/user-attachments/assets/b40d7e10-db58-4fc4-932b-d99b78c7a94d)

**Reply by mags0ft** [reply]
Oh my lord, yessss

**Comment by olumolu** [comment]
@Jeffser what you think about adding image models support i know we have talked about this before but today i just tested something i need to remove background from a image and i put that image to the new flux1 max model and it did that without any image distortion which could take me at least 30min with Photoshop or gimp so i think if we ignore the inamage and video generate stuff image models can be a really good option to have this stuff done.

**Reply by Jeffser** [reply]
Nah

**Reply by mags0ft** [reply]
Incredibly detailed answer! 👏

Now let's discuss whether we wanna add Gork support /s

**Reply by Jeffser** [reply]
you mean Grok?

**Reply by mags0ft** [reply]
Yeah, was jokingly referring to how everyone called it "Gork" in the former issues and pull requests (note the "/s") 😅

**Reply by Jeffser** [reply]
oh yeaaaahhh hahahaha

**Reply by Jeffser** [reply]
In all honesty I really hate Xitter, so I would have to think about it.

To be fair people can use `Compatible OpenAI Instances` in this case

**Reply by mags0ft** [reply]
Can totally understand - don't think about it, just don't add it at all. The model isn't worth it, and Musk's bigotry is not easily overshadowed by an LLM.

**Comment by olumolu** [comment]
what do you think about adding sambanova
https://docs.sambanova.ai/sambastudio/latest/open-ai-api.html

**Comment by nunodonato** [comment]
Hi all!
Has there been any recent discussion on enabling MCP support in Alpaca?

**Reply by mags0ft** [reply]
Yes, the maintainer does not want to implement it (??) https://github.com/Jeffser/Alpaca/issues/728

**Reply by nunodonato** [reply]
oh wow, that's sad, considering the relevance of MCP. 

I don't think doing it at the ollama level is the best. For example, LM Studio implements MCP support, and when you are chatting, you can easily toggle certain MCP's or individual tools on/off. Much like you can with the Alpaca tools. Would make total sense to make this easier for the user

**Reply by nunodonato** [reply]
> "alpaca already has lots of tooling"

 ?! Do we have the same Alpaca? I see 6 tools, 2 of which are recipe-related (!) and one is to get the date and time....

**Reply by mags0ft** [reply]
Yes, we have the same Alpaca. I don't understand it either, MCP is supremely important. Unfortunately I simply don't have the time to help implementing anything right now, so I won't be of any help currently regarding this issue.

**Reply by Jeffser** [reply]
Maintainer here, by 'lots of tooling' I meant implementations of instances, tools, model management, etc.

I think MCP is a great project but the reality is, Alpaca has pretty much everything already implemented in it's own way, moving to MCP would not only be difficult but I don't think it would benefit me and the project too much.

It's not like I'm completely against it, I just don't want to do it and I don't want the extra headache right now, but maybe in the future I **might** do it.

**Comment by mags0ft** [comment]
## 💀

<img width="1080" height="725" alt="Screenshot_20250725-011014" src="https://github.com/user-attachments/assets/db0d59a1-7c48-4db2-b6ad-ada0f7b4298b" />

**Reply by Jeffser** [reply]
Didn't even built on Flathub cause of a translation so I had to remove it lmao
